test/helpers/IntegrationHelper.js

```diff
@@ -387,6 +387,39 @@ var customMatchers = {
                 return result;
             }
         };
+    },
+    toAllBeInBottomOfDeck: function () {
+        return {
+            compare: function (cards, player, numCards) {
+                var result = {};
+                const deck = player.deck;
+                const L = deck.length;
+                result.pass = L >= numCards;
+                if (result.pass) {
+                    for (let card of cards) {
+                        result.pass = card.location === 'deck';
```

Comment:
I think as-is this means that the final test pass result will depend on what happens with the last card - shouldn't it be `result.pass = result.pass && card.location === 'deck'`? Ditto for line 411.